builtin/joined_dataset.cc

```diff
@@ -59,6 +59,30 @@ struct JoinedDataset::Itl
         //ML::compact_vector<RowHash, 2> rowHashes;   ///< Row hash from input datasets
     };
 
+     struct JoinedRowStream : public RowStream {
+
+        JoinedRowStream(JoinedDataset::Itl* source) : source(source)
+        {
+            
+        }
+
+        virtual std::shared_ptr<RowStream> clone() const{
+            auto ptr = std::make_shared<JoinedRowStream>(source);
+            return ptr;
+        }
+
+        virtual void initAt(size_t start){
+            iter = source->rows.begin() + start;
+        }
+
+        virtual RowName next() {
```

Comment:
Also, do we really need rowNames?  Or would it be better to allow the dataset to return some kind of row token, which could be used to avoid a by-name lookup when actually getting the row later on?